src/ralph/data_center/subscribers.py

```diff
@@ -141,18 +133,54 @@ def handle_update_vip_event(data):
         logger.error(msg, '; '.join(errors))
         return
 
-    ip,  = IPAddress.objects.get_or_create(address=data['ip'])
+    ip, _ = IPAddress.objects.get_or_create(address=data['ip'])
     protocol = VIPProtocol.from_name(data['protocol'].upper())
-    vip = get_vip(ip.address, data['port'], protocol.name)
+    vip = get_vip(ip, data['port'], protocol)
     if vip is None:
+        # VIP not found, should create new one.
+        return handle_create_vip_event(data)
+
+    # update cluster.
+    cluster_type, _ = ClusterType.objects.get_or_create(
+        name=data['load_balancer_type']
+    )
+    cluster, _ = Cluster.objects.get_or_create(
+        name=data['load_balancer'],
+        type=cluster_type,
+    )
+
+    if ip.ethernet.base_object != cluster:
```

Comment:
What if ip does not have `ethernet` or `ethernet` does not have a `base_object`? Some validation for that would be useful, ethernet field is not required for an IP address.

Reply:
I think we're missing two very important checks here:

1. check here if `ip.ethernet.base_object` is a cluster - we don't want to take an IP assigned to a non-cluster member and assign it to a cluster
2. check here if `ip.ethernet` is exposed in DHCP. If it is, refuse to assign it to a VIP.

Reply:
And one last thing: VIP is missing assignment of parent - see line 116 in `create_vip_event`.